### DISCUSSION TITLE
How to filter the post that titis is empty

### DISCUSSION THREAD

**Discussion by liunaijie** [hexojs/hexo#5576] [Question / Answer]
When generate post, it will read all files under `_posts` folder, I want to filter some file that not write the `title`.

I try to use `Filter` function to do this, I write a simple code like this and post it to `scripts/filterNoTitle.js`
```
hexo.extend.filter.register('before_post_render', function(data) {
    if (!data.title) {
        console.log('No title post, Skip generate file');
        return false;
    }
    return data;
}, 1);
```

When run this, it can list the file that title is empty, but It will get exception like this 

```
TypeError: Cannot read properties of undefined (reading 'includes')
    at Hexo.backtickCodeBlock
```

It looks the downstream `Filter` still run.

**Comment by stevenjoezhang** [comment]
Hexo allows posts to have empty titles, so it doesn't perform any special handling for them. If you want to hide these posts, you can set `published: false` in the Front-matter. See https://github.com/hexojs/hexo/issues/5122

**Reply by liunaijie** [reply]
Hi, thanks for the response.
Is there any posible to use `Filter` feature to do this? https://hexo.io/api/filter
When find the `title` is empty, ignore to generate the post. When I add the code I mentioned above, it looks the filter is not working, it will still run the next filteres.

You mentioned `published : false` need to add on each postes, So I want to look up is there any posible to use code to filter those postes.
